### ISSUE TITLE
model overfits

### ISSUE THREAD

**Issue by Arjun-G-Ravi** [Arjun-G-Ravi/AI-Chess#3]
Model with Smooth L1 loss, l2 and dropout regularisation, AdamW, Trapezoidal scheduler, 32 embedding, train-val = 1M, 10K, bs=10K, warmup=30
![image](https://github.com/user-attachments/assets/a9f77ebd-b374-4a33-afb2-8a464505ebf2)

Model with MSE loss, l2 and dropout regularisation, AdamW, Trapezoidal scheduler, 32 embedding, train-val = 1M, 10K, bs=10K, warmup=300

![image](https://github.com/user-attachments/assets/e043039d-70a8-45cb-b82d-46c8ccb124c5)

**Comment by Arjun-G-Ravi** [comment]
I've tried

- diff loss(mse, smooth l1)
- diff scheduler(trapezoidal, cosing, exponential)
- adjusted model size
- dataset is very big and has large variety
- changing embedding size
- added dropout, tried with many values
- AdamW has L2 regularisation inbuild, played around the parameter

No luck in preventing overfitting
